kart/sqlalchemy/adapter/postgis.py

```diff
@@ -409,23 +411,24 @@ def sql_read(self, column):
 
 
 @aliased_converter_type
-class IntervalType(ConverterType):
-    """ConverterType to that casts intervals to text - ISO8601 mode is set for durations so this does what we want."""
+class FloatType(ConverterType):
+    float32_packer = struct.Struct("@f")
 
-    def sql_read(self, column):
-        return sa.cast(column, TEXT)
+    def __init__(self, size):
+        self.size = size
+
+    def python_postread(self, value):
+        if value is not None and self.size == 32:
+            return self.float32_packer.unpack(self.float32_packer.pack(value))[0]
+        return value
 
 
 @aliased_converter_type
-class NumericType(ConverterType):
-    """ConverterType to read numerics as text. They are stored in PG as NUMERIC but we read them back as text."""
+class IntervalType(ConverterType):
+    """ConverterType to that casts intervals to text - ISO8601 mode is set for durations so this does what we want."""
```

Comment:
add an example showing the form of the result

Reply:
Done